test/resolve-ref.ts

```diff
@@ -1,64 +1,280 @@
+                });
+            }
+        });
+
+        describe('throws not found error with correct, but broken refs', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, () => {
+                    assert.rejects(
+                        () => repo.resolveRef(testCase.ref + 'n'),
+                        /ref not found refs\/heads\/mainn/
+                    );
+                });
+            }
+        });
+
+        describe('throws not found error with wrong ref', () => {
+            for (const testCase of wrongRefFixtures) {
+                it(testCase.ref, () => {
+                    assert.rejects(
+                        () => repo.resolveRef(testCase.ref + 'n'),
+                        /ref not found refs\/heads\/mainn/
+                    );
+                });
+            }
+        });
+
+        describe('shouldBeTag ref for ambiguous refs priority', () => {
+            for (const testCase of shouldBeTag) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.resolveRef(testCase.ref);
+                    const expected = testCase.oid;
+
+                    assert.strictEqual(actual, expected);
+                });
+            }
+        });
+
+        describe('shouldBeHead ref for ambiguous refs priority', () => {
+            for (const testCase of shouldBeHead) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.resolveRef(testCase.ref);
+                    const expected = testCase.oid;
+
+                    assert.strictEqual(actual, expected);
+                });
+            }
+        });
+
+        describe('shouldBeRemoteHead ref for ambiguous refs priority', () => {
+            for (const testCase of shouldBeRemoteHead) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.resolveRef(testCase.ref);
+                    const expected = testCase.oid;
+
+                    assert.strictEqual(actual, expected);
+                });
+            }
+        });
+    });
+
+    describe('expandRef()', () => {
+        it('oid into oid', async () => {
+            const oid = '7b84f676f2fbea2a3c6d83924fa63059c7bdfbe2';
+            const actual = await repo.expandRef(oid);
+
+            assert.strictEqual(actual, oid);
+        });
+
+        describe('all ref types', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.expandRef(testCase.ref);
+                    const expected = testCase.fullFormRef;
+
+                    assert.strictEqual(actual, expected);
+                });
+            }
+        });
+
+        describe('non-exists refs', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.expandRef(testCase.ref + 'n');
```

Comment:
I doubt we need to break refs here. But I see a missed test suite for invalid refs list.